projects/libs/core-library-typescript/src/base-service/service.ts

```diff
@@ -34,39 +38,23 @@ export abstract class Service {
     return Service.RPCMethod(target, propertyKey, descriptor, true);
   }
 
-  public constructor() {
+  public constructor(serviceConfig?: ServiceConfig) {
     this.name = this.constructor.name;
     this.nodeTopology = topology();
     this.peers = topologyPeers(this.nodeTopology.peers);
+    this.nodeName = (serviceConfig && serviceConfig.nodeName) || this.nodeTopology.name;
```

Comment:
I think it'd be cleaner to just require the `serviceConfig` parameter here as well.